aws-crt/Rakefile

```diff
@@ -13,7 +13,7 @@ task :compile do
   build_dir = File.expand_path('build', native_dir)
   FileUtils.mkdir_p(build_dir)
   Dir.chdir(build_dir) do
-    sh "cmake #{native_dir}" unless File.exist?('CMakeCache.txt')
+    sh "cmake #{native_dir}"
```

Comment:
I'm not actually sure whether it's safe to skip the configure step.

That said, we should mess around more with how these tasks are declare so we don't even try to build if native/ is unchanged.

Reply:
Yeah - I'd like to see how we can better define these, but I don't know really know enough about the way cmake is working here.  
Agree that its a bit unsafe and has led to confusion for me a few times - when adding new files they won't be included unless you do a `rake clean` (or delete the cache file).

Reply:
ion-c defined a script that can do all of this; our rake task could call something similar defined in aws-crt

https://github.com/amzn/ion-c/blob/master/build-release.sh

Reply:
Yeah - I do think we should pull some of this out - atleast to avoid the duplication w/ the extconf.rb.... but we can address that separately.